File: LeetCode/5회차/B12-121-Best Time to Buy and Sell Stock.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        low = 10000
        res = 0
        for price in prices:
            res = max(price - low, res)
            if price < low:
                low = price
        return res


print(Solution().maxProfit([7,1,5,3,6,4]))

File: LeetCode/JavaScript/B12-121-Best Time to Buy and Sell Stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
  let low = Infinity,
    res = 0;
  for (price of prices) {
    res = Math.max(price - low, res);
    low = Math.min(price, low);
  }
  return res;
};

console.log(maxProfit([7, 1, 5, 3, 6, 4]));
